modules/redis/scanner.go

```diff
@@ -228,44 +361,84 @@ func (scanner *Scanner) Scan(target zgrab2.ScanTarget) (zgrab2.ScanStatus, inter
 	}
 	defer scan.Close()
 	result := scan.result
-	pingResponse, err := scan.SendCommand("PING")
+	pingResponse, err := scan.SendCommand(scanner.commandMappings["PING"])
 	if err != nil {
-		// if the first command fails (as opposed to succeeding but returning an
+		// If the first command fails (as opposed to succeeding but returning an
 		// ErrorMessage response), then flag the probe as having failed.
 		return zgrab2.TryGetScanStatus(err), nil, err
 	}
 	// From this point forward, we always return a non-nil result, implying that
 	// we have positively identified that a redis service is present.
 	result.PingResponse = forceToString(pingResponse)
 	if scanner.config.Password != "" {
-		authResponse, err := scan.SendCommand(scanner.config.AuthCommand, scanner.config.Password)
+		authResponse, err := scan.SendCommand(scanner.commandMappings["AUTH"], scanner.config.Password)
 		if err != nil {
 			return zgrab2.TryGetScanStatus(err), result, err
 		}
 		result.AuthResponse = forceToString(authResponse)
 	}
-	infoResponse, err := scan.SendCommand("INFO")
+	infoResponse, err := scan.SendCommand(scanner.commandMappings["INFO"])
 	if err != nil {
 		return zgrab2.TryGetScanStatus(err), result, err
 	}
 	result.InfoResponse = forceToString(infoResponse)
-	infoResponseBulk, ok := infoResponse.(BulkString)
-	if ok {
+	if infoResponseBulk, ok := infoResponse.(BulkString); ok {
 		for _, line := range strings.Split(string(infoResponseBulk), "\r\n") {
-			if strings.HasPrefix(line, "redis_version:") {
-				result.Version = strings.SplitN(line, ":", 2)[1]
-				break
+			line_prefix_suffix := strings.SplitN(line, ":", 2)
+			if len(line_prefix_suffix) == 1 {
+				line_prefix_suffix = append(line_prefix_suffix, "")
+			}
```

Comment:
Comment, not a request for change: Rather than needing to do an append operation here, you could make this a bit simpler and just:

```
prefix := line_prefix_suffix[0]
var suffix string
if len(line_prefix_suffix) == 1 {
   suffix = ""
} else {
   suffix = line_prefix_suffix[1]
}
```

And just switch on `prefix` and assign / convert `suffix` as appropriate.

But this isn't some critical code path for performance, so no need to change it to optimize away a single `append` call (and since the append almost certainly won't need to reallocate for a single `""`, it probably only saves a function call or so).

Reply:
Changed to this, however slightly changed the logic since a string's zero value is an empty string